tests/ae-inventory-container-itest/src/test/java/org/metaeffekt/core/itest/container/AlpineTest.java

```diff
@@ -60,6 +61,15 @@ public void analyse() throws Exception {
         Assert.assertTrue(AbstractCompositionAnalysisTest.testSetup.rebuildInventory());
     }
 
+    @Test
+    public void testComponentPatterns() throws Exception {
+        final Inventory inventory = AbstractCompositionAnalysisTest.testSetup.getInventory();
+        final Inventory referenceInventory = AbstractCompositionAnalysisTest.testSetup.readReferenceInventory();
+        final File baseDir = new File(AbstractCompositionAnalysisTest.testSetup.getScanFolder());
+        boolean hasDuplicates = ComponentPatternValidator.removeDuplicateComponentPatterns(referenceInventory, inventory, baseDir);
```

Comment:
Can we change this to a signature to deliver the deviations; such that we not just get a boolean, but can work with the results.
I expect that in some cases we have duplicates, but we need to qualify them as acceptable.

Reply:
we agreed on making a complex object to make assertions and complex logic possible, so that we can define what we expect and also make some exceptions to match our results